src/backend/app/Services/AgentSoldApplianceService.php

```diff
@@ -97,4 +109,97 @@ function ($q) use ($agentId) {
             )->latest()
             ->paginate();
     }
+
+    public function getAgentsByCustomerId(int $customerId): Collection {
+        return Agent::whereHas('soldAppliances', function ($query) use ($customerId) {
+            $query->where('person_id', $customerId);
+        })->get();
+    }
+
+    public function processSale(AgentSoldAppliance $agentSoldAppliance, array $requestData = []): void {
```

Comment:
For some consistency with the rest of the code, could we call this function `processSaleFromRequest`?

Reply:
Updated.